app/containers/TextInput/FormTextInput.tsx

```diff
@@ -58,112 +57,91 @@ export interface IRCTextInputProps extends TextInputProps {
 	containerStyle?: StyleProp<ViewStyle>;
 	inputStyle?: StyleProp<TextStyle>;
 	inputRef?: React.Ref<RNTextInput>;
-	testID?: string;
 	iconLeft?: TIconsName;
 	iconRight?: TIconsName;
 	left?: JSX.Element;
-	onIconRightPress?(): void;
-	theme: TSupportedThemes;
 }
 
-interface IRCTextInputState {
-	showPassword: boolean;
-}
-
-export default class FormTextInput extends React.PureComponent<IRCTextInputProps, IRCTextInputState> {
-	static defaultProps = {
-		error: {},
-		theme: 'light'
-	};
-
-	state = {
-		showPassword: false
-	};
+const FormTextInput = React.memo(
+	({
+		label,
+		error,
+		loading,
+		containerStyle,
+		inputStyle,
+		inputRef,
+		iconLeft,
+		iconRight,
+		left,
+		testID,
+		secureTextEntry,
+		placeholder,
+		...inputProps
+	}: IRCTextInputProps) => {
+		const { colors, theme } = useTheme();
+		const [showPassword, setShowPassword] = useState(false);
 
-	get iconLeft() {
-		const { testID, iconLeft, theme } = this.props;
-		return iconLeft ? (
-			<CustomIcon
-				name={iconLeft}
-				testID={testID ? `${testID}-icon-left` : undefined}
-				size={20}
-				color={themes[theme].bodyText}
-				style={[styles.iconContainer, styles.iconLeft]}
-			/>
-		) : null;
-	}
+		const showIconLeft = () =>
+			iconLeft ? (
+				<CustomIcon
+					name={iconLeft}
+					testID={testID ? `${testID}-icon-left` : undefined}
+					size={20}
+					color={colors.bodyText}
+					style={[styles.iconContainer, styles.iconLeft]}
+				/>
+			) : null;
 
-	get iconRight() {
-		const { iconRight, onIconRightPress, theme } = this.props;
-		return iconRight ? (
-			<Touchable onPress={onIconRightPress} style={[styles.iconContainer, styles.iconRight]}>
-				<CustomIcon name={iconRight} size={20} color={themes[theme].bodyText} />
-			</Touchable>
-		) : null;
-	}
+		const showIconRight = () =>
```

Comment:
You should prevent the usage of this kind of function creation inside function components.
You're only recreating something that React can't really understand.
On this case, it's actually easy to just abstract and reuse it inside the return.